webui/src/components/community_admin.jsx

```diff
@@ -0,0 +1,316 @@
+import { fromJS, OrderedMap, Map } from 'immutable';
+import { Wait, Err } from './waiting.jsx';
+import Immutable from 'immutable';
+
+export const CommunityAdmin = React.createClass({
+    mixins: [React.addons.PureRenderMixin],
+
+    // Assign a role to a user
+    handleChange(roles, email, selectedRole){
+        var searchedUser = {};
+        if(email && selectedRole ){
+             serverCache.registerUserRole(email, roles[selectedRole], 
+                    () => {
+                        notifications.success("The new role was added to the user"); 
+                    },
+                    () => {
+                        notifications.danger("An error occured while trying to add the new role to the user. ");
+                    }    
+                );
+        }
+    },
+
+    render() {
+        const communityName = this.props.location.pathname.replace(new RegExp("/communities/|/communityadminpage",'g'),"");
+        const communitiesListTemp = serverCache.getCommunities();
+        if (communitiesListTemp instanceof Error) {
+            return <Err err={communitiesListTemp}/>;
+        }
+        var communitiesList = communitiesListTemp.reduce(function(map, community){
+                map[community.get('id').replace(new RegExp("-",'g'),"")] = community.get('name');
+            return map;
+        });
+        // Check if the community name in the url is an existing community, if not show an error and do nothing!
+        if(communitiesList){
+            var tag = true;
+            Object.keys(communitiesList).forEach( community => {
+                if(communitiesList[community] === communityName) {
+                    tag = false;
+                }
+            })
+            if(tag){
+                return (
+                    <div className="alert-danger"> <h3>{communityName} community doesn't exist </h3> </div> 
+                );
+            }}
+
+        var communityID = null;
+        for(var key in communitiesList) {
+            if(communitiesList[key] === communityName) {
+                communityID = key
+            }
+        }
+
+        var roles = {};
+        if(communityID){
+            roles = serverCache.getCommunityRoles(communityID);
+        }
+
+        if (roles instanceof Error) {
+            return <Err err={roles}/>;
+        }
+        var roles_dict = {};
+        if(roles.size > 0){
+            roles.map( role => {
+                roles_dict[role.getIn(['name']).replace(new RegExp("^com:[^:]*:"), "")] = role.getIn(['id'])
+            })
+        }
+
+        var users = {};
+        if(roles.size > 0 ){ 
+            roles.forEach( role => { 
+                users[role.get('id')] = serverCache.getCommunityUsers(role.getIn(['id']));
+            })
+        }
+
+        var mergedUsers = {}; 
+        Object.keys(users).forEach(role => { 
+            if(typeof(users[role]) !== "undefined" && Object.keys(users[role]).length > 0 ){
```

Comment:
`if (users[role])` doesn't seem to be sufficient. When the `users` doesn’t have value yet, it’s either an empty object, or `undefined`.